src/Common/serde.h

```diff
@@ -27,6 +27,27 @@ void ALWAYS_INLINE deserialize(S & s, RB & rb, VersionType version, Args &&... a
     s.deserialize(rb, version, std::forward<Args>(args)...);
 }
 
+/// With owned versions
+template <typename S, typename WB, typename... Args>
+concept Serializable
+    = requires(const S & s, WB & wb, Args &&... args) { s.serialize(wb, std::forward<Args>(args)...); };
+
+template <typename S, typename RB, typename... Args>
+concept Deserializable
+    = requires(S & s, RB & rb, Args &&... args) { s.deserialize(rb, std::forward<Args>(args)...); };
+
+template <typename WB, typename... Args, Serializable<WB, Args...> S>
+void ALWAYS_INLINE serialize(const S & s, WB & wb, Args &&... args)
+{
+    s.serialize(wb, std::forward<Args>(args)...);
+}
+
+template <typename RB, typename... Args, Deserializable<RB, Args...> S>
+void ALWAYS_INLINE deserialize(S & s, RB & rb, Args &&... args)
```

Comment:
I actually don't understand why we need this since calling s.deserialize / serialize on sites are more readable than yet another forwarding / layer : both at code layer and machine code layer and for debugging purpose

Reply:
I would highly suggest writing direct / simple code without possible perf regression.

Reply:
Yes, it is actually useless. It is only used to unify the code style.